x/delayedack/client/cli/query.go

```diff
@@ -84,20 +79,33 @@ func CmdGetPacketsByRollapp() *cobra.Command {
 			req := &types.QueryRollappPacketsRequest{
 				RollappId: rollappId,
 				Status:    commontypes.Status_PENDING, // get pending packets by default
+				Type:      commontypes.Type_UNDEFINED, // must specify, as '0' is a valid type
 			}
 
 			if len(args) > 1 {
-				// Use the map to convert a string to an enum
 				statusStr := strings.ToUpper(args[1])
-				status, ok := statusMap[statusStr]
+				status, ok := commontypes.Status_value[statusStr]
 				if !ok {
 					// Handle error: statusStr is not a valid commontypes.Status
 					return fmt.Errorf("invalid status: %s", statusStr)
 				}
-				req.Status = status
+				req.Status = commontypes.Status(status)
+			}
+
+			if len(args) > 2 {
+				typeStr := strings.ToUpper(args[2])
```

Comment:
As I understand in the code, in order to filter by type, we need to filter it by status first. We should make the code separate between status and type.

Reply:
in `packets-by-rollapp` the status and type are optional filters. In `packets-by-status` the type is optional. But you still have `packets-by-type` so you can use that to filter by type alone, granted the status will be PENDING by default.

Reply:
What I mean is like, when you assert like that, the filter must always be status first, and then after that it can be type. So it's kinda annoying if I want to filter just by type